pkg/operator/ceph/cluster/cluster.go

```diff
@@ -160,13 +160,20 @@ func (c *cluster) validateCephVersion(version *cephver.CephVersion) error {
 		return nil
 	}
 
-	if c.Spec.External.Enable {
+	if c.Spec.External.Enable && c.Spec.External.RunStatelessWorkload {
 		c.Info.CephVersion, err = cephspec.ValidateCephVersionsBetweenLocalAndExternalClusters(c.context, c.Namespace, *version)
 		if err != nil {
 			return fmt.Errorf("failed to validate ceph version between external and local. %+v", err)
 		}
 	}
 
+	// On external cluster setup, if we don't bootstrap any resources in the Kubernetes cluster then
+	// there is no need to validate the Ceph image further
+	if c.Spec.External.Enable && !c.Spec.External.RunStatelessWorkload {
+		logger.Debug("external cluster not allowing the bootstrap of other CRs (like rgw, mds or nfs), not validating Ceph version.")
```

Comment:
Reading this message doesn't quite explain to me what Rook is doing. I think maybe referencing `runStatelessWorkload` here being the reason bootstrap isn't allowed would be good.